libmamba/CMakeLists.txt

```diff
@@ -264,6 +264,22 @@ set(
     ${LIBMAMBA_SOURCE_DIR}/api/shell.cpp
     ${LIBMAMBA_SOURCE_DIR}/api/update.cpp
 )
+# TODO: remove when switch to C++20
+if(CMAKE_CXX_COMPILER_ID STREQUAL "Clang" OR CMAKE_CXX_COMPILER_ID STREQUAL "AppleClang")
+    # This file uses capturing structured bindings, which was fixed in C++20
+    set_source_files_properties(
+        ${LIBMAMBA_SOURCE_DIR}/download/mirror_impl.cpp
+        PROPERTIES COMPILE_FLAGS -Wno-c++20-extensions
```

Comment:
There is nothing wrong with our code and nothing to change there. So we can safely ignore the warning we're using `C++20` feature (capturing structured binding).
This can't hide any real problems, so here this warning/error will disappear.

Reply:
Technically, this construction is C++20, not C++17. It works with gcc and msvc though, and the warning is still here with Clang, even if we pass the C++20 option (that's a known bug), so we can definitely hide this one.